pkg/build/api/validation/validation.go

```diff
@@ -64,14 +64,15 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) field.ErrorList {
 	fromRefs := map[string]struct{}{}
 	specPath := field.NewPath("spec")
 	triggersPath := specPath.Child("triggers")
+	buildFrom := buildutil.GetImageStreamForStrategy(config.Spec.Strategy)
```

Comment:
Anything that tries to update the bc spec would be broken - do any of the
controllers invoke PUT buildconfigs?

On Feb 1, 2016, at 6:48 PM, Ben Parees <notifications@github.com> wrote:

In pkg/build/api/validation/validation.go
<https://github.com/openshift/origin/pull/6952#discussion_r51500952>:

> @@ -64,14 +64,15 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) field.ErrorList {
>  	fromRefs := map[string]struct{}{}
>  	specPath := field.NewPath("spec")
>  	triggersPath := specPath.Child("triggers")
> +	buildFrom := buildutil.GetImageStreamForStrategy(config.Spec.Strategy)

depends...when would that etcd content go back through validation? certain
"oc edit bc" could be broken (they'd have to fix the BC before it would
save).

Is there any other time?

if so then yes, any BC that has a default ICT but does not have the right
"From" type would fail validation when previously it did not (but
previously the ICT had no effect).

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6952/files#r51500952>.